include/circt/Dialect/RTL/RTLTypes.h

```diff
@@ -32,6 +32,13 @@ bool isRTLIntegerType(mlir::Type type);
 /// dialects.
 bool isRTLValueType(mlir::Type type);
 
+/// Return the hardware bit width of a type. Does not reflect any encoding,
+/// padding, or storage scheme, just the bit (and wire width) of a
+/// statically-size type. Reflects the number of wires needed to transmit a
+/// value of this type. Returns -1 if the type is not known or cannot be
+/// statically computed.
+int getBitWidth(mlir::Type type);
```

Comment:
I'd recommend returning int64_t so we can support really big hardware :-)

Reply:
Larger sizes are also useful for modeling really wide data structures which don't end up as hardware that wide. E.g. they get serialized.